frame/balances/src/lib.rs

```diff
@@ -1056,6 +1056,71 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		});
 		Ok(actual)
 	}
+
+	fn do_transfer(
```

Comment:
Shouldn't currency also use this instead of repeating code?

Reply:
Currency wants a `DispatchResult `to be returned, while this returns `Result<T::Balance, DispatchError>`.
I can see how to convert it once returned.

Reply:
yeah